usaspending_api/search/delta_models/subaward_search.py

```diff
@@ -539,6 +564,58 @@
         global_temp.ref_country_code AS rcc
             ON (rcc.country_code = UPPER(bs.sub_legal_entity_country_code)
                 AND bs.sub_legal_entity_country_code IS NOT NULL)
+    -- Congressional District '90' represents multiple congressional districts
+    LEFT OUTER JOIN
+        global_temp.cd_state_grouped pop_cd_state_grouped ON (
+            pop_cd_state_grouped.state_abbreviation=UPPER(bs.sub_place_of_perform_state_code)
+            AND pop_cd_state_grouped.congressional_district_no <> '90'
+        )
+    LEFT OUTER JOIN
+        global_temp.cd_state_grouped rl_cd_state_grouped ON (
+            rl_cd_state_grouped.state_abbreviation=UPPER(bs.sub_legal_entity_state_code)
+            AND rl_cd_state_grouped.congressional_district_no <> '90'
+        )
+    LEFT OUTER JOIN
+        raw.zips pop_zips ON (
+            pop_zips.zip5=LEFT(COALESCE(sub_place_of_performance_zip, ''), 5)
+            AND pop_zips.zip_last4=RIGHT(COALESCE(sub_place_of_performance_zip, ''), 4)
+        )
+    LEFT OUTER JOIN
+        raw.zips rl_zips ON (
+            rl_zips.zip5=LEFT(COALESCE(bs.sub_legal_entity_zip, ''), 5)
+            AND rl_zips.zip_last4=RIGHT(COALESCE(bs.sub_legal_entity_zip, ''), 4)
+        )
+    LEFT OUTER JOIN
+        global_temp.cd_zips_grouped pop_cd_zips_grouped ON (
+            pop_cd_zips_grouped.zip5=LEFT(COALESCE(sub_place_of_performance_zip, ''), 5)
+            AND pop_cd_zips_grouped.state_abbreviation=UPPER(bs.sub_place_of_perform_state_code)
+        )
+    LEFT OUTER JOIN
+        global_temp.cd_zips_grouped rl_cd_zips_grouped ON (
+            rl_cd_zips_grouped.zip5=LEFT(COALESCE(bs.sub_legal_entity_zip, ''), 5)
+            AND rl_cd_zips_grouped.state_abbreviation=UPPER(bs.sub_legal_entity_state_code)
+        )
+    LEFT OUTER JOIN
+        global_temp.cd_city_grouped pop_cd_city_grouped ON (
+            pop_cd_city_grouped.city_name=UPPER(bs.sub_place_of_perform_city_name)
+            AND pop_cd_city_grouped.state_abbreviation=UPPER(bs.sub_place_of_perform_state_code)
+        )
+    LEFT OUTER JOIN
+        global_temp.cd_city_grouped rl_cd_city_grouped ON (
+            rl_cd_city_grouped.city_name=UPPER(bs.sub_legal_entity_city_name)
+            AND rl_cd_city_grouped.state_abbreviation=UPPER(bs.sub_legal_entity_state_code)
+        )
+    LEFT OUTER JOIN
+        global_temp.cd_county_grouped pop_cd_county_grouped ON (
+            pop_cd_county_grouped.county_number=LPAD(CAST(CAST(REGEXP_EXTRACT(pop.county_numeric, '^[A-Z]*(\\d+)(?:\\.\\d+)?$', 1) AS SHORT) AS STRING), 3, '0')
```

Comment:
My rule of thumb is to keep comments to explaining why. I think there was less knowledge duplication before you added the comment.

> This ticket has nothing to do with that question
Can you explain why this story has nothing to do with my question when you've added new occurrences here?

> It's how we've always extracted congressional district code from the field (get the two digits, zero pad it).

So you're saying that because it's how we've always done it that any new place we do it more doesn't mean new code should follow best practices.

My original comment was suggesting you to add an explanation of why you are doing these transforms, that's it.

Reply:
So this was how we extracted congressional district from the field long ago and I'm simply keeping that logic when matching on fields. I could add a comment along the lines of "congressional district fields may include characters digits, so we're extracting the digits and making sure it's still zero padded" but that would be a guess. If we wanted to actually get to the root of that question, we'd need to analyze each raw field in production data to see its values looking for any edge cases and then determine if the transformation logic is necessary/needs to be updated everywhere, which is not in the scope.